crates/fuel-core/src/schema.rs

```diff
@@ -99,7 +103,7 @@ where
     //  It means also returning `has_previous_page` and `has_next_page` values.
     // entries(start_key: Option<DBKey>)
     F: FnOnce(&Option<SchemaKey>, IterDirection) -> StorageResult<Entries>,
-    Entries: Iterator<Item = StorageResult<(SchemaKey, SchemaValue)>>,
+    Entries: Stream<Item = StorageResult<(SchemaKey, SchemaValue)>>,
```

Comment:
nit: Somewhat unrelated since the function used to take `Iterator` and not `IntoIterator`; It's more flexible to allow `IntoStream` here - unless there is a particular reason to limit this?

```suggestion
    Entries: IntoStream<Item = StorageResult<(SchemaKey, SchemaValue)>>,
```

Reply:
`IntoStream` doesn't exist =)

Reply:
Okay yeah I was thinking about `https://docs.rs/futures/0.3.31/futures/future/struct.IntoStream.html` but we're using the tokio trait here.

Reply:
It is structure, not a trait

Reply:
It is not weird that the `IntoStream` trait doesn't exist for streams. The structure `IntoStream` exists to bypass compiler and convert `TryStream` into `Stream`

Reply:
A `TryStream` is already a `Stream` though, but I guess this explicit conversion can be useful as you describe. The example in the docs is to convert a future to a stream, but it feels circular to me because if the future already implements `StreamExt` you basically have all the stream functionality for it already so why would you need to call `into_stream()` explicitly in that case?

It still feels a bit weird to me, but I think the similarity with the `IntoIterator` name throws me off a bit here.